GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -79,6 +79,9 @@ private bool TryDownloadUpgrade(out string errorMessage)
                 if (newerVersion == null)
                 {
                     // Already up-to-date
+                    // Make sure there a no asset installers remaining in the Downloads directory. This can happen if user
+                    // upgraded by manually downloading and running asset installers.
+                    productUpgrader.DeletePreviousDownloads();

```

Comment:
What happens if this fails? What happens if it throws?

Reply:
Added exception tracing in ```DeletePreviousDownloads()``` method. This is not treated as a critical error, upgrade would log the error and proceed.